version: '3.7'

services:
  app:
    image: chaosdorf/pawebcontrol:latest
    environment:
      - PULSE_SERVER=pulse.chaosdorf.space
    networks:
      - traefik
    deploy:
      mode: replicated
      replicas: 1
      restart_policy:
        delay: 60s
        max_attempts: 5
      resources:
        limits:
          cpus: '0.50'
          memory: 256M
        reservations:
          cpus: '0.25'
          memory: 8M
      labels:
        - traefik.docker.network=traefik_net
        - traefik.port=8000
        - traefik.frontend.headers.STSSeconds=2592000
        - traefik.frontend.rule=Host:pulseweb.chaosdorf.space,pulseweb
        - traefik.frontend.errors.network.backend=error-app
        - traefik.frontend.errors.network.status=401,403,404,429,500,503
        - traefik.frontend.errors.network.query=/{status}.html

networks:
  traefik:
    driver: overlay
    external: true
    name: traefik_net
